book/src/spec.md

```diff
@@ -0,0 +1,628 @@
+
+> The identifier is case-insensitive because compatibility with VHDL is
+> desired.
+
+#### Bitcount
+
+The bit count cannot be zero.
+
+> A lot of synthesis tools out there do not handle null ranges very well.
+
+### Number of element lanes (N)
+
+The data described by the element content is physically replicated this many
+times, allowing multiple elements to be transferred per stream handshake.
+\\(N\\) must be an integer greater than or equal to one.
+
+### Dimensionality (D)
+
+This parameter specifies how many `last` bits are needed to represent the data.
+It must be an integer greater than or equal to zero.
+
+> Intuitively, each list nesting level adds one to this number. For instance,
+> to stream two-dimensional lists, two last bits will be needed: one to mark
+> the boundaries of the inner list, and one to mark the boundary of each
+> two-dimensional list.
+
+### Complexity (C)
+
+This parameter specifies what guarantees the source makes about how elements
+are transferred. Equivalently, it specifies how many assumptions the sink can
+make.
+
+| Higher C                        | Lower C                        |
+|---------------------------------|--------------------------------|
+| Fewer guarantees made by source | More guarantees made by source |
+| Source is easier to implement   | Source is harder to implement  |
+| Sink can make fewer assumptions | Sink can make more assumptions |
+| Sink is harder to implement     | Sink is easier to implement    |
+
+The complexity levels and signals are defined such that a source with
+complexity \\(C_a\\) can be connected to a sink with complexity
+\\(C_b \ge C_a\\).
+
+The complexity number is defined as a nonempty, period-separated list of
+nonnegative integers. A complexity number is higher than another when the
+leftmost integer is greater, and lower when the leftmost integer is lower. If
+the leftmost integer is equal, the next integer is checked recursively. If one
+complexity number has more entries than another, the shorter number is padded
+with zeros on the right.
+
+> That is, it works a bit like a version number. The idea behind this is to
+> allow future version of this specification (or any other kind of addition
+> later) to insert new complexity levels between the ones defined here. For
+> instance, 3 < 3.1 < 3.1.1 < 3.2 < 4.
+
+The restrictions imposed by certain complexity levels are listed in detail
+along with the unconditional signal requirements in later sections.
+Intuitively, however, the complexity number carries the following significance.
+
+| \\(C\\) | Description                                                                         |
+|---------|-------------------------------------------------------------------------------------|
+| < 8     | Only one sequence can be terminated per transfer.                                   |
+| < 7     | Which data lanes are enabled can be described with a simple range.                  |
```

Comment:
Those two sentences mean different things, the suggestion is wrong (a set of enabled data lanes is never described by just a simple range, but also by element type, N, and so on). Changed to "The indices of the active data lanes can be described with a simple range." (also changed "enabled" to "active" for consistency because I use the latter later on)